src/schnorr.rs

```diff
@@ -33,38 +31,63 @@ pub type UntweakedPublicKey = XOnlyPublicKey;
 #[derive(Clone, Copy, PartialEq, Eq, PartialOrd, Ord, Hash, Debug)]
 pub struct TweakedPublicKey(XOnlyPublicKey);
 
-/// A trait for tweaking Schnorr public keys
+/// A trait for tweaking Schnorr key types (x-only public keys and key pairs).
 pub trait TapTweak {
```

Comment:
In 063a8f5fffe90e8bd729c10b254825dbb4420076:

Why do we even have this trait? Why not use the one from Bitcoin?

Reply:
Same with this entire module, really.